<a href="https://colab.research.google.com/github/NEEL5252/Global-Data-Analysis/blob/main/Analyzing_of_the_global_terrorism_dataset.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **PROJECT NAME**: Global Terrorism Dataset

### **PROJECT TYPE:** EDA
### **CONTRIBUTOR:** Individual

# **PROJECT SUMMARY**

Using the Global Terrorism Database (GTD) from 1970 to 2017, this research investigates essential facets of global terrorism through extensive analysis. The findings include locations with various assault intensities, trends in annual attack frequency, success rates, and the top nations with both the highest and lowest fatalities. It dives into the number of injuries received in each nation, the number of fatalities and injuries per area, the weapons used, and the geographical predominance of assault-based assaults. This study strengthens policymaking, counterterrorism efforts, and academic research by providing greater insights into the varied processes of terrorism. Finally, it prepares stakeholders to improve global security measures, creating a proactive approach to effectively battling terrorism.


# GITHUB link:


https://github.com/NEEL5252

# Problem Statement:

1. Attacks overall by areas
2. Total assaults for the year, whether they went down or up.
3. The number of successful and unsuccessful assaults.
4. The top 10 nations in terms of both kill rates and kills per capita
5. The total number of victims injured in terrorist acts worldwide
6. The overall death toll by geographic region
7. The overall number of victims per location of injuries
8. The total number of weapons used 9. The location of all the assaults that took place.


# Buisseness Objective

# **LET'S START THE ANALYSIS**

## **PHASE - 1**


```
#  Importing libraries, importing dataset and read the dataset
```



In [2]:
# Importing necessary libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import plotly.express as px
import seaborn as sns
import plotly.graph_objects as go

In [3]:
# Mount the google drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
# Load the csv file / Read the csv file
globalTd = pd.read_csv("/content/drive/MyDrive/Global Terrarisom dataSet/Global Terrorism Data.csv", encoding='ISO-8859-1')

<ipython-input-4-6f2e09c616af>:2: DtypeWarning: Columns (4,6,31,33,61,62,63,76,79,90,92,94,96,114,115,121) have mixed types. Specify dtype option on import or set low_memory=False.
  globalTd = pd.read_csv("/content/drive/MyDrive/Global Terrarisom dataSet/Global Terrorism Data.csv", encoding='ISO-8859-1')


In [5]:
# Print the globalTd data
globalTd.head()

eventid  iyear  imonth  iday approxdate  extended resolution  country  \
0  197000000001   1970       7     2        NaN         0        NaN       58   
1  197000000002   1970       0     0        NaN         0        NaN      130   
2  197001000001   1970       1     0        NaN         0        NaN      160   
3  197001000002   1970       1     0        NaN         0        NaN       78   
4  197001000003   1970       1     0        NaN         0        NaN      101   

          country_txt  region  ... addnotes scite1 scite2  scite3  dbsource  \
0  Dominican Republic       2  ...      NaN    NaN    NaN     NaN      PGIS   
1              Mexico       1  ...      NaN    NaN    NaN     NaN      PGIS   
2         Philippines       5  ...      NaN    NaN    NaN     NaN      PGIS   
3              Greece       8  ...      NaN    NaN    NaN     NaN      PGIS   
4               Japan       4  ...      NaN    NaN    NaN     NaN      PGIS   

   INT_LOG  INT_IDEO INT_MISC INT_ANY  related  
0        0         0        0       0      NaN  
1        0         1        1       1      NaN  
2       -9        -9        1       1      NaN  
3       -9        -9        1       1      NaN  
4       -9        -9        1       1      NaN  

[5 rows x 135 columns]

In [6]:
# Shape of the globalTd dataset
globalTd.shape

(181691, 135)

In [7]:
# Count the null and missing values from the particular column of the dataset
nullcounts = globalTd.isnull().sum()
# Total null and missing counts from the dataset
totalNull = nullcounts.sum()
print(totalNull)

13853997


### Insights of the dataset:-


Together, there are 181691 rows and columns. 1,38,53,997 pieces of information are either missing or null in total. The year, kind, quantity of fatalities, nature of the weapon used, quantity of injuries, and city of the assault are among the details included in this data collection. Null values and missing values for several columns need to be removed. The wrangling and cleansing of the data will soon be finished.

